crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -278,13 +278,15 @@ def upgrade_controller_nodes
         return false unless non_founder_api.disable_pre_upgrade_attribute_for founder.name
 
         # 3. delete old pacemaker resources
-        # FIXME: couldn't we do it right after migrating l3 routers from the first node?
         delete_pacemaker_resources non_founder.name
 
-        # 4. start crowbar-join at the first node
+        # 4. migrate routers from nodes being upgraded
+        return false unless founder_api.router_migration
```

Comment:
I think we need to check either here or in ```router_migration``` if the node really has the "neutron-network" role. And skip calling the script if it doesn't.